src/redux/todosSlice.ts

```diff
@@ -64,8 +68,16 @@ const todosSlice = createSlice({
         todo.completed = !todo.completed;
       }
     },
+    addMany: (state, action: PayloadAction<Todo[]>) => {
+      state.todos.push(...action.payload);
+    },
   },
 });
 
+export const loadTodosAsync = (): AppThunk => async (dispatch) => {
+  const loadedState = await loadState();
+  dispatch(todosSlice.actions.addMany(loadedState.todos));
```

Comment:
If I understood this right, you are initializing your state here. In that case you should create and use `setTodos` action that will replace anything there is in the store with your content